new-components/targets/git-clone/docker-compose.yaml

```diff
@@ -0,0 +1,30 @@
+services:
+  target:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    platform: linux/amd64
+    env_file:
+      - .env.git-clone
+    volumes:
+      - ./testing/testdata/testrepo:/workspace
+    depends_on:
+      gitea:
+        condition: service_healthy
+  gitea:
+    image: gitea/gitea:latest
```

Comment:
Setting up a local GitHub like backend to be used for local setup and testing https://about.gitea.com/.